src/ape_run/_cli.py

```diff
@@ -76,13 +76,15 @@ def invoke(self, ctx: Context) -> Any:
             if ctx.params["interactive"]:
                 # Print the exception trace and then launch the console
                 # Attempt to use source-traceback style printing.
-                if not isinstance(err, ApeException) or not handle_ape_exception(
-                    err, [ctx.obj.project_manager.path]
-                ):
-                    err_info = traceback.format_exc()
-                    click.echo(err_info)
-
-                self._launch_console()
+                network_value = ctx.params.get("network") or networks.default_ecosystem.name
+                with networks.parse_network_choice(network_value, disconnect_on_exit=False):
```

Comment:
**idea**: We could add a comment here like "NOTE: We must be connected to the provider before launching the console so that the active provider is set from the last context.